### PR TITLE
Fix/get s value

### PR DESCRIPTION
**Pull request by hakuturu583**
## Types of PR

- [ ] New Features
- [ ] Upgrade of existing features
- [x] Bugfix

## Description

Fix problem in getSValue function, this function allowed to return s values under 0 or over 1.
Remove unnecessary parameter "threshold_distance".

## How to review this PR.

Please check passing all CI test cases.

### CODE REVIEW

File: simulation/traffic_simulator/include/traffic_simulator/math/catmull_rom_spline.hpp [hunk 1]
```diff
@@ -39,9 +39,8 @@ class CatmullRomSpline
   const std::vector<geometry_msgs::msg::Point> getTrajectory(
     double start_s, double end_s, double resolution) const;
   boost::optional<double> getSValue(
-    geometry_msgs::msg::Point position, double threshold_distance = 3.0,
-    unsigned int initial_resolution = 30, unsigned int max_iteration = 30,
-    double tolerance = 0.001);
+    geometry_msgs::msg::Point position, unsigned int initial_resolution = 30,
+    unsigned int max_iteration = 30, double tolerance = 0.03);
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/math/catmull_rom_spline.hpp:43]: `unsigned int` => `std::size_t`

https://stackoverflow.com/questions/131803/unsigned-int-vs-size-t

File: simulation/traffic_simulator/include/traffic_simulator/math/hermite_curve.hpp [hunk 2]
```diff
@@ -57,11 +57,10 @@ class HermiteCurve
   double getLength(size_t num_points) const;
   double getLength() const { return length_; }
   boost::optional<double> getSValue(
-    geometry_msgs::msg::Point position, double threshold_distance = 1.0,
-    unsigned int initial_resolution = 30, unsigned int max_iteration = 30, double tolerance = 0.001,
-    bool autoscale = false) const;
+    const geometry_msgs::msg::Point & point, unsigned int initial_resolution = 30,
+    unsigned int max_iteration = 30, double tolerance = 0.03, bool autoscale = false) const;
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/math/hermite_curve.hpp:61]: `unsigned int` => `std::size_t`

https://stackoverflow.com/questions/131803/unsigned-int-vs-size-t

File: simulation/traffic_simulator/include/traffic_simulator/math/hermite_curve.hpp [hunk 3]
```diff
@@ -72,8 +71,11 @@ class HermiteCurve
 private:
   std::pair<double, double> get2DMinMaxCurvatureValue() const;
   double length_;
+  boost::optional<double> solveNewtonMethod(
+    const geometry_msgs::msg::Point & point, double initial_value, double tolerance,
+    unsigned int max_iteration) const;
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/math/hermite_curve.hpp:76]: `unsigned int` => `std::size_t`

https://stackoverflow.com/questions/131803/unsigned-int-vs-size-t

File: simulation/traffic_simulator/src/math/catmull_rom_spline.cpp [hunk 4]
```diff
@@ -315,15 +315,15 @@ boost::optional<double> CatmullRomSpline::getCollisionPointIn2D(
 }
 
 boost::optional<double> CatmullRomSpline::getSValue(
-  geometry_msgs::msg::Point position, double threshold_distance, unsigned int initial_num_points,
-  unsigned int max_iteration, double tolerance)
+  geometry_msgs::msg::Point position, unsigned int initial_num_points, unsigned int max_iteration,
+  double tolerance)
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/src/math/catmull_rom_spline.cpp:319]: `unsigned int` => `std::size_t`

https://stackoverflow.com/questions/131803/unsigned-int-vs-size-t

File: simulation/traffic_simulator/src/math/hermite_curve.cpp [hunk 5]
```diff
@@ -163,55 +163,48 @@ boost::optional<double> HermiteCurve::getCollisionPointIn2D(
   return *std::min_element(s_values.begin(), s_values.end());
 }
 
+boost::optional<double> HermiteCurve::solveNewtonMethod(
+  const geometry_msgs::msg::Point & point, double initial_value, double tolerance,
+  unsigned int max_iteration) const
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/src/math/hermite_curve.cpp:168]: `unsigned int` => `std::size_t`

https://stackoverflow.com/questions/131803/unsigned-int-vs-size-t

File: simulation/traffic_simulator/src/math/hermite_curve.cpp [hunk 6]
```diff
@@ -163,55 +163,48 @@ boost::optional<double> HermiteCurve::getCollisionPointIn2D(
...
+      return boost::none;
+    }
+  }
+  return ret;
+}
+
 boost::optional<double> HermiteCurve::getSValue(
-  geometry_msgs::msg::Point point, double threshold_distance, unsigned int initial_resolution,
+  const geometry_msgs::msg::Point & point, unsigned int initial_resolution,
   unsigned int max_iteration, double tolerance, bool autoscale) const
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/src/math/hermite_curve.cpp:186]: `unsigned int` => `std::size_t`

https://stackoverflow.com/questions/131803/unsigned-int-vs-size-t

File: simulation/traffic_simulator/src/math/hermite_curve.cpp [hunk 7]
```diff
@@ -163,55 +163,48 @@ boost::optional<double> HermiteCurve::getCollisionPointIn2D(
...
-    std::min_element(initial_errors.begin(), initial_errors.end());
-  size_t index = std::distance(initial_errors.begin(), iter);
-  ret = initial_value_candidates[index];
   std::vector<double> errors;
   std::vector<double> s_values;
-  for (unsigned i = 0; i < max_iteration; i++) {
-    double error = getSquaredDistanceIn2D(point, ret);
-    if (std::fabs(error) < (tolerance * tolerance)) {
-      return ret;
+  for (unsigned int i = 0; i < initial_resolution; i++) {
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/src/math/hermite_curve.cpp:192]: `unsigned int` => `std::size_t`

https://stackoverflow.com/questions/131803/unsigned-int-vs-size-t

**Review by yamacir-kit — CHANGES_REQUESTED** [review]

File: simulation/traffic_simulator/src/math/hermite_curve.cpp (hunk 7 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/math/hermite_curve.cpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/603/commits/0b3461ec444280b0c20ed2b8e5b008d42e74cc5e

File: simulation/traffic_simulator/src/math/hermite_curve.cpp (hunk 6 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/math/hermite_curve.cpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/603/commits/0b3461ec444280b0c20ed2b8e5b008d42e74cc5e

File: simulation/traffic_simulator/src/math/hermite_curve.cpp (hunk 5 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/math/hermite_curve.cpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/603/commits/0b3461ec444280b0c20ed2b8e5b008d42e74cc5e

File: simulation/traffic_simulator/src/math/catmull_rom_spline.cpp (hunk 4 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/math/catmull_rom_spline.cpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/603/commits/71e2f34e3d6f0fa135e9708cba12d90aa0421824

File: simulation/traffic_simulator/include/traffic_simulator/math/hermite_curve.hpp (hunk 3 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/math/hermite_curve.hpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/603/commits/71e2f34e3d6f0fa135e9708cba12d90aa0421824

File: simulation/traffic_simulator/include/traffic_simulator/math/hermite_curve.hpp (hunk 2 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/math/hermite_curve.hpp]: fixed in https://github.com/tier4/scenario_simulator_v2/commit/0b3461ec444280b0c20ed2b8e5b008d42e74cc5e

File: simulation/traffic_simulator/include/traffic_simulator/math/catmull_rom_spline.hpp (hunk 1 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/math/catmull_rom_spline.hpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/603/commits/71e2f34e3d6f0fa135e9708cba12d90aa0421824

**Review by yamacir-kit — APPROVED** [review]

**Pull request merged**